### ISSUE TITLE
Partial Active Patterns with arguments are flagged with RulesNamingConventionsUnderscoreError

### ISSUE THREAD

**Issue by inosik** [fsprojects/FSharpLint#357]
### Description

Partial active patterns are flagged with the underscore naming rule, if they have arguments. The suggested change to fix the error produces invalid code.

### Repro steps

Change this:

``` fsharp
let (|Empty|_|) = function
    | "" -> Some Empty
    | _ -> None
```

To this:

``` fsharp
let (|Empty|_|) str =
    match str with
    | "" -> Some Empty
    | _ -> None
```

### Expected behavior

Partial active patterns should not be flagged by this rule, 

### Actual behavior

The suggested change is this:

``` diff
-let (|Empty|_|) str =
+let (|Empty||) str =
     match str with
     | "" -> Some Empty
     | _ -> None
```

I guess FSharpLint misinterprets this as an active pattern with multiple choices, instead of a partial active pattern.

### Known workarounds

For simple cases, use a pattern matching function. Otherwise live with the errors.

### Related information

Version 0.12.1 of FSharpLint.Core, shipped by Ionide.

**Comment by jgardella** [comment]
Thanks for the bug report!
